### PR TITLE
Optionally put short struct variants on one line

### LINKED ISSUE #418: Consider putting struct enum variants on one line
```
-    Switch { adt_def: AdtDef<'tcx> },
+    Switch {
+        adt_def: AdtDef<'tcx>,
+    },
 
     // test for equality
-    Eq { value: Literal<'tcx>, ty: Ty<'tcx> },
+    Eq {
+        value: Literal<'tcx>,
+        ty: Ty<'tcx>,
+    },
 
     // test whether the value falls within an inclusive range
-    Range { lo: Literal<'tcx>, hi: Literal<'tcx>, ty: Ty<'tcx> },
+    Range {
+        lo: Literal<'tcx>,
+        hi: Literal<'tcx>,
+        ty: Ty<'tcx>,
+    },
```

I'm not sure if the original is better or not, tbh, We should have an option for this at the least.

### PR DESCRIPTION
**Pull request by nrc**
Closes #418

r? @marcusklaas

### CODE REVIEW

File: src/items.rs [hunk 1]
```diff
@@ -758,7 +761,8 @@ fn format_struct_struct(context: &RewriteContext,
                         fields: &[ast::StructField],
                         generics: Option<&ast::Generics>,
                         span: Span,
-                        offset: Indent)
+                        offset: Indent,
+                        one_line_width: Option<usize>)
```
Comment (marcusklaas) [inline src/items.rs]: Is this ever called with `None`? Or future proofing?

Comment (nrc) [inline src/items.rs]: Regular structs call it with `None` (c.f., variant structs)

File: src/items.rs [hunk 2]
```diff
@@ -813,23 +817,35 @@ fn format_struct_struct(context: &RewriteContext,
...
         ends_with_newline: true,
         config: context.config,
     };
-    Some(format!("{}\n{}{}\n{}}}",
-                 result,
-                 offset.block_indent(context.config).to_string(context.config),
-                 try_opt!(write_list(items, &fmt)),
-                 offset.to_string(context.config)))
+    let items_str = try_opt!(write_list(&items, &fmt));
+    if one_line_width.is_some() && !items_str.contains('\n') {
```
Comment (marcusklaas) [inline src/items.rs]: Turbo nit: it may be cleaner to match on `tactic`, but w/e.

**Comment by marcusklaas** [comment]
Nice!

**Pull request merged**